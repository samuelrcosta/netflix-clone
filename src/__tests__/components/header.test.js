import { render, fireEvent } from '@testing-library/react';
import { Header } from '../../components';
import { MemoryRouter } from 'react-router-dom';
import * as ROUTES from '../../constants/routes';

describe('<Header />', () => {
  it('renders basic <Header /> with a default background', () => {
    const { container, getByText, getByTestId, queryByTestId } = render(
      <MemoryRouter>
        <Header>
          <Header.Frame>
            <Header.Logo to={ROUTES.HOME} src="./logo.svg" alt="Netflix" />
            <Header.ButtonLink to={ROUTES.SIGN_IN}>Sign In</Header.ButtonLink>
          </Header.Frame>
        </Header>
      </MemoryRouter>
    );

    expect(getByText('Sign In')).toBeTruthy();
    expect(getByTestId('header-bg')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the fulll <Header /> ith a background', () => {
    const { container, getByText, getByTestId } = render(
      <MemoryRouter>
        <Header src="joker1" dontShowOnSmallViewPort>
          <Header.Frame>
            <Header.Group>
              <Header.Logo to={ROUTES.HOME} src="/images/logo.svg" alt="Netflix" />
              <Header.TextLink active={'false'} onClick={() => {}}>
                Series
              </Header.TextLink>
              <Header.TextLink active onClick={() => {}}>
                Films
              </Header.TextLink>
            </Header.Group>
            <Header.Group>
              <Header.Search searchTerm="Joker" setSearchTerm={() => {}} />
              <Header.Profile>
                <Header.Picture src="/images/user.png" alt="User" />
                <Header.Dropdown>
                  <Header.Group>
                    <Header.Picture src="/images/user.png" alt="User" />
                    <Header.TextLink>User Test</Header.TextLink>
                  </Header.Group>
                  <Header.Group>
                    <Header.TextLink onClick={() => {}}>Sign out</Header.TextLink>
                  </Header.Group>
                </Header.Dropdown>
              </Header.Profile>
            </Header.Group>
          </Header.Frame>
          <Header.Feature>
            <Header.FeatureCallOut>Watch Joker Now</Header.FeatureCallOut>
            <Header.Text>Forever alone in a crowd</Header.Text>
            <Header.PlayButton>Play</Header.PlayButton>
          </Header.Feature>
        </Header>
      </MemoryRouter>
    );

    expect(getByTestId('search-input')).toBeTruthy();
    expect(getByTestId('search-input').value).toBe('Joker');
    fireEvent.change(getByTestId('search-input'), {
      target: { value: 'Simpsons' },
    });
    fireEvent.click(getByTestId('search-click'));
    expect(getByText('Series')).toBeTruthy();
    expect(getByText('Films')).toBeTruthy();
    expect(getByText('User Test')).toBeTruthy();
    expect(getByText('Watch Joker Now')).toBeTruthy();
    expect(getByText('Sign out')).toBeTruthy();
    expect(getByText('Forever alone in a crowd')).toBeTruthy();
    expect(getByText('Play')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders basic <Header /> without a background', () => {
    const { container, getByText, getByTestId, queryByTestId } = render(
      <MemoryRouter>
        <Header bg={false}>
          <Header.Frame>
            <Header.Logo to={ROUTES.HOME} src="./logo.svg" alt="Netflix" />
            <Header.ButtonLink to={ROUTES.SIGN_IN}>Sign In</Header.ButtonLink>
          </Header.Frame>
        </Header>
      </MemoryRouter>
    );

    expect(getByText('Sign In')).toBeTruthy();
    expect(queryByTestId('header-bg')).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
