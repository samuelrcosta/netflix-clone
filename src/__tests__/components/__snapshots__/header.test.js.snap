// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders basic <Header /> with a default background 1`] = `
<div
  class="header__Background-djltcn-0 iOPNDS"
  data-testid="header-bg"
>
  <div
    class="header__Container-djltcn-1 kWSTgG"
  >
    <a
      href="/"
    >
      <img
        alt="Netflix"
        class="header__Logo-djltcn-2 hxjAVL"
        src="./logo.svg"
      />
    </a>
    <a
      class="header__Button-djltcn-4 tZDcd"
      href="/signin"
    >
      Sign In
    </a>
  </div>
</div>
`;

exports[`<Header /> renders basic <Header /> without a background 1`] = `
<div
  class="header__Container-djltcn-1 kWSTgG"
>
  <a
    href="/"
  >
    <img
      alt="Netflix"
      class="header__Logo-djltcn-2 hxjAVL"
      src="./logo.svg"
    />
  </a>
  <a
    class="header__Button-djltcn-4 tZDcd"
    href="/signin"
  >
    Sign In
  </a>
</div>
`;

exports[`<Header /> renders the fulll <Header /> ith a background 1`] = `
<div
  class="header__Background-djltcn-0 jLMYsP"
  data-testid="header-bg"
  src="joker1"
>
  <div
    class="header__Container-djltcn-1 kWSTgG"
  >
    <div
      class="header__Group-djltcn-3 kXnqMt"
    >
      <a
        href="/"
      >
        <img
          alt="Netflix"
          class="header__Logo-djltcn-2 hxjAVL"
          src="/images/logo.svg"
        />
      </a>
      <p
        class="header__Link-djltcn-9 dHFOOS"
      >
        Series
      </p>
      <p
        class="header__Link-djltcn-9 dHFOOS"
      >
        Films
      </p>
    </div>
    <div
      class="header__Group-djltcn-3 kXnqMt"
    >
      <div
        class="header__Search-djltcn-13 YTpUK"
      >
        <button
          class="header__SearchIcon-djltcn-14 bkRkrB"
          data-testid="search-click"
        >
          <img
            alt="Search"
            src="/images/icons/search.png"
          />
        </button>
        <input
          class="header__SearchInput-djltcn-15 bmTGpn"
          data-testid="search-input"
          placeholder="Search films and series"
          value="Joker"
        />
      </div>
      <div
        class="header__Profile-djltcn-12 gagDWK"
      >
        <button
          alt="User"
          class="header__Picture-djltcn-10 ckKQVC"
          src="/images/users//images/user.png.png"
        />
        <div
          class="header__Dropdown-djltcn-11 kCVvbj"
        >
          <div
            class="header__Group-djltcn-3 kXnqMt"
          >
            <button
              alt="User"
              class="header__Picture-djltcn-10 ckKQVC"
              src="/images/users//images/user.png.png"
            />
            <p
              class="header__Link-djltcn-9 dHFOOS"
            >
              User Test
            </p>
          </div>
          <div
            class="header__Group-djltcn-3 kXnqMt"
          >
            <p
              class="header__Link-djltcn-9 dHFOOS"
            >
              Sign out
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="header__Container-djltcn-1 header__Feature-djltcn-5 kWSTgG dCYvwl"
  >
    <h2
      class="header__FeatureCallOut-djltcn-6 jcGnnM"
    >
      Watch Joker Now
    </h2>
    <p
      class="header__Text-djltcn-7 dKXRKA"
    >
      Forever alone in a crowd
    </p>
    <button
      class="header__PlayButton-djltcn-8 bgCFLp"
    >
      Play
    </button>
  </div>
</div>
`;
